src/Aspire.Hosting/Dashboard/DashboardViewModelService.cs

```diff
@@ -39,6 +40,7 @@ internal sealed partial class DashboardViewModelService : IDashboardViewModelSer
     private readonly Dictionary<(ResourceKind, string), List<string>> _resourceAssociatedServicesMap = [];
     private readonly ConcurrentDictionary<string, List<EnvVar>> _additionalEnvVarsMap = [];
     private readonly HashSet<string> _containersWithTaskStarted = [];
+    private readonly HashSet<string> _checkedServices = [];
```

Comment:
Change to `ConcurrentDictionary<string, object?>`. Multiple threads will get/set values.

There isn't a concurrent set so the work around is to use concurrent dictionary and ignore the value (set `new object()` or `null`)